### PR TITLE
Fix browsing by product tag

### LINKED ISSUE #2626: [Marketplace] Receive error when browsing by Tag
### Expected behavior
No errors

### Actual Behavior
Receive error in console
```
modules.js?hash=6d2e3df…:84778 Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the TagNavContainer component.
```

### Steps to Reproduce the Behavior
1. Create a new product
1. Create a new tag
1. Add tag to product
1. Click on tag in nav
1. Observe error

**Note** This takes a little bit of work to replicate. It doesn't happen every time but if you flip back and forth between tags and then wait a second or two on the added tag page you can see the error.

### Versions
```
Node: 7.10.0
NPM: 5.0.3
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: marketplace
Docker: 17.06.0-ce
commit: b6e18a4b5a35b852bd75648ae395a2f4b77eae81
```

### PR DESCRIPTION
**Pull request by foladipo**
Resolves #2626.

This PR fixes a bug in the `marketplace` branch. The cause was that this app was executing a certain callback that does `setState` in a React component that had been unmounted. This caused an error to be thrown in the browser console.

## Test instructions
1. Create a new product.
1. Create a few (new) tags to that product. 
1. Complete the creation of that product (by filling other fields) and publish it.
1. Add one of these tags to the list of main tags in the navbar. (They are highlighted in the screenshot below.)
1. Then, add two or more tags as sub-tags of the first.
1. Navigate to the root of the app e.g `localhost:3000`.
1. Click the sub tag created above to go to `localhost:3000/tags/SUB_TAG` and click outside the navbar.
1. Check the browser console. It should be free of the error reported [here](https://github.com/reactioncommerce/reaction/issues/2626).
1. Repeat the last three steps a few more times using other sub-tags/tags.

highlighted - screen shot 2017-09-06 at 6 42 33 pm

### CODE REVIEW

File: imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js [hunk 1]
```diff
@@ -114,6 +116,7 @@ const wrapComponent = (Comp) => (
     componentDidMount() {
       window.addEventListener("resize", this.onWindowResize);
       this.onWindowResize();
+      this._isMounted = true;
```
Comment (zenweasel) [inline imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js]: What is prefacing this variable with an underscore supposed to indicate/accomplish?

File: imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js [hunk 2]
```diff
@@ -259,11 +263,13 @@ const wrapComponent = (Comp) => (
       this.setState({ attachedBodyListener: false });
     }
 
+    // TODO: If this is a method of TagNavContainer, why is it, and many
+    // others, declared as an arrow function?
```
Comment (zenweasel) [inline imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js]: I don't know if a TODO code comment is the best place to get a programming question answered.

**Comment by zenweasel** [comment]
This works and I would like to merge it but I don't know what's up with the underscored variables

File: imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js (hunk 1 above)
Comment (foladipo) [reply imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js]: I prefixed that variable with an underscore to show that it's a custom implementation of the now deprecated `ReactComponent.isMounted()` as advised in [this official React post](https://facebook.github.io/react/blog/2015/12/16/ismounted-antipattern.html).

Comment (foladipo) [reply imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js]: Furthermore, it's consistent with the `_unmounted` variable in `imports/plugins/core/components/lib/composer/compose.js`.

Comment (zenweasel) [reply imports/plugins/core/ui-tagnav/client/containers/tagNavContainer.js]: Great. Works for me.

**Pull request merged**